packages/fxa-settings/src/pages/Signin/container.tsx

```diff
@@ -90,32 +84,19 @@ import { MetricsContext } from 'fxa-auth-client/browser';
  * `/signup` to match content-server functionality.
  */
 
-function getAccountInfo(nonCachedEmail?: string) {
-  let email = nonCachedEmail;
+function getAccountInfo(email?: string) {
   let sessionToken: hexstring | undefined;
   let uid: hexstring | undefined;
 
-  const storedLocalAccount = currentAccount();
-  // Try to use local storage values if email is not provided or if email
-  // is provided and matches the email in local storage
-  if (!nonCachedEmail || storedLocalAccount?.email === nonCachedEmail) {
-    sessionToken = storedLocalAccount?.sessionToken;
-    email = storedLocalAccount?.email;
-    uid = storedLocalAccount?.uid;
-  }
-
-  // If there is still not an email, try locating the last logged in account.
-  if (!email) {
-    const lastStoredLocalAccount = lastStoredAccount();
-    if (lastStoredLocalAccount) {
-      email = lastStoredLocalAccount.email;
-      sessionToken = lastStoredLocalAccount.sessionToken;
-      uid = lastStoredLocalAccount.uid;
-      setCurrentAccount(lastStoredLocalAccount.uid);
+  if (email) {
```

Comment:
What if `email` is not provided, don't we want to try to read from localStorage in that case?

Reply:
I would suggest we let the email first page be sole account chooser and revisit when we convert the email first page to React (otherwise we should directly check in fxa-settings if the user is enrolled in the experiment vs if the flag is turned on).

Reply:
> I would suggest we let the email first page be sole account chooser and revisit when we convert the email first page to React (otherwise we should directly check in [fxa-settings](https://mozilla-hub.atlassian.net/browse/FXA-settings) if the user is enrolled in the experiment vs if the flag is turned on).

Unless I'm misunderstanding, why would we need to check for the experiment status to check local storage? It seems like we could have that there for either case (if the user is in the experiment vs not). I am not if we want to disallow users to reach `/signin` directly, but I guess that's more of a UX decision.

Reply:
I just realized we have very similar logic (and the same named function) in `ConfirmSignupCode`. Not sure this warrants any changes, just wanted to point it out.

Reply:
Look into the caching issue some more... the `useMutation` hook is triggering a re-render of the component, and causing `getAccountInfo` to be called again before completing the sign in. When the local storage data is modified between page load and page submission, the `email` argument might change or become undefined if email was retrieved from local storage vs query param.

Reply:
Discussed with @LZoog that this fix is acceptable for now to address the bug, but we may want to re-enable directly checking on the signin page for an account in local storage, with mitigation to avoid re-rendering the page when the `useMutation` hook is used. Filed: https://mozilla-hub.atlassian.net/browse/FXA-9624

Suggested working on this change once React signin is fully rolled out and we are working on converting the email-first page.